libs/ledger/src/chain/main_chain.cpp

```diff
@@ -806,11 +813,20 @@ void MainChain::RecoverFromFile(Mode mode)
 
     if (in.is_open())
     {
-      byte_array::ByteArray bloom_filter_data{in};
+      try
+      {
+        byte_array::ByteArray bloom_filter_data{in};
 
-      LargeObjectSerializeHelper buffer{bloom_filter_data};
+        LargeObjectSerializeHelper buffer{bloom_filter_data};
 
-      buffer >> bloom_filter_;
+        buffer >> bloom_filter_;
+      }
+      catch (std::exception &e)
+      {
+        FETCH_LOG_ERROR(LOGGING_NAME,
+                        "Failed to load Bloom filter from storage! Reason: ", e.what());
+        bloom_filter_.Reset();
```

Comment:
There's a problem here, because now the main chain won't have a Bloom filter, so any transaction that comes up will be treated as new even if it is already in the chain. At the moment I think this requires a reset of the chain, as well as the filter.